lib/cache.js

```diff
@@ -8,8 +8,26 @@ const VERSION = require('../package').version;
 
 const TimeoutError = bluebird.TimeoutError;
 
-function createCacheKey(segment, id) {
+function createCacheKey(segment, ctx, opts) {
   const versionedSegment = `http-transport:${VERSION}:${segment}`;
+  const requiredIdPart = `${ctx.req.getMethod()}:${ctx.req.getUrl()}`;
+  let id = requiredIdPart;
+  const varyOn = _.get(opts, 'varyOn');
+
+  if (varyOn) {
+    const headers = ctx.req.getHeaders();
+    const optionalIdPart = [];
+
+    varyOn.forEach((key) => {
```

Comment:
Probably better to use a map here as returning from a map is expected. e.g 

Something like:
```js
  const optionalIdPart = varyOn.map((key) => {
    if (headers[key]) {
      return `${key}=${headers[key]}`;
    }
    return `${key}=`;
  });
```